### PR TITLE
Back link on bitstream edit page

### PR DESCRIPTION
**Pull request by jonas-atmire**
## References
This leverages the following 2 links:
* Link to [REST Contract](https://github.com/atmire/Rest7Contract/commit/1932830c82d3a303db542abaeb482754d40937ec) - Bundle to Item 
* Link to [REST Contract](https://github.com/DSpace/Rest7Contract/blob/master/bitstreams.md) - Bitstream to Bundle 

## Description
At the moment, when you are on the edit bitstreams page, and you click the back button, Angular redirects you to the previous page based on the browsing history.
For example:
* When first going to the homepage, and then going to the bitstream edit page through a URL, and not when going through the intermediate edit page steps, the back button goes to the home page again.
* When accessing that page after first visiting google, the back button simply redirects back to google
* When accessing that from an external program, the back buttons simply doesn't work at all since no 'previous' location is know.

This should actually go to the edit item page on the bitstream tab (Similar to how it’s done in previous DSpace versions and on other item edit pages)

## Instructions for Reviewers
Please add a more detailed description of the changes made by your PR. At a minimum, providing a bulleted list of changes in your PR is helpful to reviewers.

List of changes in this PR:
* On the bitstream/bundle objects, the relevant exposed links, as described in the 2 mentioned rest contracts have been added as HALLinks.
* On the bitstream edit page, these links are resolved to retrieve the owning bundle, and subsequently the owning item to retrieve the relevant information to link back to the required page.

**Include guidance for how to test or review your PR.** 
As a test, go through the different steps mentioned in the description.
When clicking the 'cancel' button, all of these steps should return the user to the edit item page, on the bitstream tab

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Review by tdonohue — COMMENTED** [review]
@jonas-atmire : I gave this a quick review & test today.

The code overall looks fine.  But, this doesn't seem to be working as described.  That said, I'm not sure if it's a fault of this PR or not.  Here's what I'm seeing (I built this PR using `yarn start`):

1. Login as an Admin
2. Edit an Item
3. Click on Bitstreams tab, then click Cancel
4. I'm returned to the Item page (*not* the edit Item screen)

That said, I see that same behavior if I also click on the "Metadata" tab or "Relationship" tab on the Edit Item screen. Clicking cancel on any of them redirects me back to the Item page (`/items/[uuid]`), not the Edit Item page (`/items/[uuid]/edit`).

I also just realized I can reproduce this on the demo site (https://dspace7-demo.atmire.com/) for the "Metadata" or "Relationships" tab.  So, it doesn't look like this PR is causing that bug.

In summary, I think this PR has improved the behavior of the "Bitstreams" tab, as it now redirects to the same location as every other tab when clicking "Cancel".  However, none of those tabs seems to redirect to the "Edit Item" page as I would expect when clicking "Cancel".   Could you look into whether we can get all these tabs redirecting to the right place? Maybe it's a relatively simple fix.

**Comment by jonas-atmire** [comment]
@tdonohue I think there might have been a slight miscommunication.
The 4 points you mention there are not exactly what is being altered. Please refer to these, slightly altered steps to reproduce the fix:

1. Login as an Admin
2. Edit an Item
3. Click on Bitstreams tab, then edit a bitstream
4. Click cancel
5. You should be redirected to the edit item screen on the bitstream tab

**Comment by tdonohue** [comment]
Hi @jonas-atmire : Apologies for the delay in retesting this.  You are correct, I had misunderstood your prior instructions.

After further testing, I'm able to verify the bug & that this PR fixes the issue.  I was previously testing from the wrong page (I was clicking cancel on the "Edit Item" page previously, and not the "Edit Bitstream" page).

So, this looks good to me! 👍

**Review by tdonohue — APPROVED** [review]
